src/main/java/buildcraftAdditions/proxy/ClientProxy.java

```diff
@@ -59,4 +62,8 @@ public EntityPlayer getPlayer(String name) {
 	public void cloneFluidTextures(Fluid source, Fluid target) {
 		target.setIcons(source.getStillIcon(), source.getFlowingIcon());
 	}
+
+	public static GuiScreen getCurrentOpenedScreen() {
+		return Minecraft.getMinecraft().currentScreen;
```

Comment:
this needs to be static as the CuiScreen object doesn't exist on the server, so it will crash the commonproxy, just adding this here as it's the best place to put it in my opninion